### PR TITLE
Relay multiple paths from the same process

### PR DESCRIPTION
**Pull request by agouin**
Wires up event processor multiple `PathProcessors` for a single `ChainProcessor` and adds accompanying `ibctest` case and github workflow.

image

Usage:
```
rly start path-1 path-2 path-n
```

### CODE REVIEW

File: relayer/strategies.go [hunk 1]
```diff
@@ -31,42 +30,55 @@ const (
...
+				dst: processor.NewPathEnd(pathName, p.Dst.ChainID, p.Dst.ClientID, filter.Rule, filterDst),
+			})
+		}
+
+		go relayerStartEventProcessor(ctx, log, chainProcessors, ePaths, initialBlockHistory, maxTxSize, maxMsgLength, memo, errorChan, metrics)
 		return errorChan
 	case ProcessorLegacy:
-		go relayerMainLoop(ctx, log, src, dst, filter, maxTxSize, maxMsgLength, memo, errorChan)
+		if len(paths) != 1 {
+			panic(fmt.Errorf("only one path supported for legacy processor. pass `-p events` for multiple paths"))
```
Comment (jtieri) [inline relayer/strategies.go:75]: #989 switches to the event processor as the default relayer behavior so maybe this error message needs to be changed slightly? I would imagine at that point explicitly using `-p events` is no longer needed and something like `-p l` would be used to switch to the legacy processor

File: ibctest/relay_many_test.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+func TestRelayerMultiplePathsSingleProcess(t *testing.T) {
+	relayeribctest.BuildRelayerImage(t)
+
+	client, network := ibctest.DockerSetup(t)
+	r := ibctest.NewBuiltinRelayerFactory(
+		ibc.CosmosRly,
+		zaptest.NewLogger(t),
+		ibctestrelayer.CustomDockerImage(relayeribctest.RelayerImageName, "latest", "100:1000"),
+		ibctestrelayer.ImagePull(false),
+		ibctestrelayer.StartupFlags("-p", "events"),
```
Comment (jtieri) [inline ibctest/relay_many_test.go:32]: This line could be dropped after #989 is merged

**Review by jtieri — APPROVED** [review]
Great work! Everything looks relatively straightforward.

File: relayer/strategies.go (hunk 1 above)
Comment (agouin) [reply relayer/strategies.go:75]: will include this in #989

File: ibctest/relay_many_test.go (hunk 2 above)
Comment (agouin) [reply ibctest/relay_many_test.go:32]: will include this in #989

**Pull request merged**